File: backend/src/main/java/com/festago/domain/TicketRepository.java
package com.festago.domain;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface TicketRepository extends JpaRepository<Ticket, Long> {

    List<Ticket> findAllByStageId(Long stageId);

    Optional<Ticket> findByTicketTypeAndStage(TicketType ticketType, Stage stage);

    @Query("""
        SELECT t FROM Ticket t
        JOIN FETCH t.stage s
        JOIN FETCH t.ticketEntryTimes et
        WHERE t.id = :ticketId
        """)
    Optional<Ticket> findByIdWithFetch(@Param("ticketId") Long ticketId);
}

File: android/festago/app/src/main/java/com/festago/festago/domain/repository/TicketRepository.kt
package com.festago.festago.domain.repository

import com.festago.festago.domain.model.Ticket
import com.festago.festago.domain.model.TicketCode

interface TicketRepository {
    suspend fun loadTicket(ticketId: Long): Result<Ticket>
    suspend fun loadCurrentTickets(): Result<List<Ticket>>
    suspend fun loadTicketCode(ticketId: Long): Result<TicketCode>
    suspend fun loadHistoryTickets(size: Int): Result<List<Ticket>>
}
